components/event-sources/reconciler/httpsource/controller.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2019 The Kyma Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package httpsource implements a controller for the HTTPSource custom resource.
+package httpsource
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"k8s.io/client-go/tools/cache"
+
+	messaginginformersv1alpha1 "knative.dev/eventing/pkg/client/injection/informers/messaging/v1alpha1/channel"
+	"knative.dev/eventing/pkg/reconciler"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/resolver"
+	servingclient "knative.dev/serving/pkg/client/injection/client"
+	knserviceinformersv1alpha1 "knative.dev/serving/pkg/client/injection/informers/serving/v1alpha1/service"
+
+	sourcesv1alpha1 "github.com/kyma-project/kyma/components/event-sources/apis/sources/v1alpha1"
+	sourcesclient "github.com/kyma-project/kyma/components/event-sources/client/generated/injection/client"
+	httpsourceinformersv1alpha1 "github.com/kyma-project/kyma/components/event-sources/client/generated/injection/informers/sources/v1alpha1/httpsource"
+)
+
+const (
+	// reconcilerName is the name of the reconciler
+	reconcilerName = "HTTPSources"
+
+	// controllerAgentName is the string used by this controller to identify
+	// itself when creating events.
+	controllerAgentName = "http-source-controller"
+
+	// adapterImageEnvVar is the name of the environment variable containing the
+	// container image of the HTTP receive adapter.
+	adapterImageEnvVar = "HTTP_ADAPTER_IMAGE"
+)
+
+// NewController returns a new controller that reconciles HTTPSource objects.
+func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+	httpSourceInformer := httpsourceinformersv1alpha1.Get(ctx)
+	knServiceInformer := knserviceinformersv1alpha1.Get(ctx)
+	chInformer := messaginginformersv1alpha1.Get(ctx)
+
+	rb := reconciler.NewBase(ctx, controllerAgentName, cmw)
+	r := &Reconciler{
+		Base:             rb,
+		adapterImage:     getAdapterImage(),
```

Comment:
Wouldn't it make sense to read the adapter image from the configmap as well?
If we support multiple controllers in the manager we would also need multiple images in the environment variables and changing one of them would require a restart of the manager. With a configmap watcher we could update the image at runtime...

Reply:
I agree with you, but shall we wait until this use cases materializes? The PR is humongous already, adding more background logic for something that we don't even use yet sounds a little bit off-topic to me for now.

Reply:
sounds valid